src/main/java/com/ciaranmckenna/readinglistapp/service/AuthorServiceImpl.java

```diff
@@ -0,0 +1,75 @@
+package com.ciaranmckenna.readinglistapp.service;
+
+import com.ciaranmckenna.readinglistapp.dao.entity.Author;
+import com.ciaranmckenna.readinglistapp.dao.repository.AuthorRepository;
+import com.ciaranmckenna.readinglistapp.dto.AuthorRecord;
+import com.ciaranmckenna.readinglistapp.exceptions.NotFoundException;
+import com.ciaranmckenna.readinglistapp.util.Mapper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class AuthorServiceImpl implements AuthorService {
+
+    private final AuthorRepository authorRepository;
+
+    @Override
+    public List<AuthorRecord> findAllAuthors() {
+        return authorRepository.findAll()
+                .stream()
+                .map(Mapper::mapAuthorEntityToAuthorRecord)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<AuthorRecord> findByAuthorNameContainingIgnoreCase(String firstName, String lastName) {
+        List<Author> authors;
+        if (firstName != null && !firstName.isEmpty()) {
+            // Search by first name
+            authors = authorRepository.findByFirstNameContainingIgnoreCase(firstName);
+        } else if (lastName != null && !lastName.isEmpty()) {
+            // Search by last name
+            authors = authorRepository.findByLastNameContainingIgnoreCase(lastName);
+        } else {
+            // If neither first name nor last name is provided, return all authors
+            return Collections.emptyList();
```

Comment:
I am not sure this logic is doing what we want it to do. 
We should be returning a complete list of authors if the entry to the name fields is blank.
A return of Collections.emptyList() returns an empty list. 

Perhaps we should return a call to findAllAuthors(); as it will return the full list of authors

Reply:
regarding logic, you need an empty list there. so you can use this, or return simply authors (the one from 34 ...) because there are 3 cases and one, when there is no param! if you dont return empty list will return full list, no matter if it finds or not by firstName or lastName (but the main focus was params are not required ...)